File: ruby/easy/repeated_substring_pattern.rb
# @param {String} s
# @return {Boolean}
def repeated_substring_pattern(s)
    len = 1
    while len <= s.length/2
        if((s.length % len) == 0)
            i, j = 0, len
            while j < s.length
                break if(s[i] != s[j])
                i += 1 and j += 1
            end
            return true if j == s.length
        end
        len += 1
    end
    return false
end

File: javascript/easy/repeatedSubstringPattern.js
/**
 * @param {string} s
 * @return {boolean}
 */
var repeatedSubstringPattern = function(s) {
    let len = 1;
    while(len <= parseInt(s.length/2)) {
        if((s.length % len) === 0) {
            let i = 0, j = i+len, repeats = true;
            while(j < s.length) {
                if(s[i] !== s[j]) {
                    repeats = false;
                    break;
                }
                i++; j++;
            }
            if(repeats) {
                return true;
            }
        }
        len++;
    }
    return false;
};
